functions/src/fetch_friend_versions.ts

```diff
@@ -0,0 +1,85 @@
+import admin from "./firebase";
+import * as functions from "firebase-functions";
+import * as cors from "cors";
+import { apiError } from "./api";
+
+const friendsCollection = admin.firestore().collection("friends") as FirebaseFirestore.CollectionReference<FriendsData>;
+const auth = admin.auth();
+const schedulesCollection = admin.firestore().collection("schedules-dev");
+
+const corsHandler = cors({ origin: true });
+
+interface FriendsData {
+  terms: Record<string, Record<string, string[]>>[];
+}
+
+type VersionReturn = Record<string, string[]>;
```

Comment:
I haven't been able to use the endpoint due to the proposed changes, but I'm a bit confused as to why this is only `Record<string, string[]>`. Aren't we fetching a schedule of the following type?

![image](https://user-images.githubusercontent.com/38119460/219982111-b206e8aa-52b4-47ef-a294-da6b5c0ceb1e.png)

Reply:
yea, my bad